doc/copp/CoPP Config and Management.md

```diff
@@ -0,0 +1,111 @@
+
+# Requirements
+
+The following are the high level requirements for CoPP/TRAP managment
+
+1. CoPP Tables shall be part of the Config DB. Currently this is only present in APP DB
+2. User shall be able to configure, modify CoPP table entries for various policing requirements
+3. System shall be backward compatible with old Config DB without any CoPP Tables
+4. A VS test must be added to verify the Config DB CoPP tables
+
+# Design Proposal
+
+## Current behaviour
+
+During SWSS start, the prebuilt [copp.json](https://github.com/Azure/sonic-swss/blob/201911/swssconfig/sample/00-copp.config.json) file is loaded as part of start script [swssconfig.sh](https://github.com/Azure/sonic-buildimage/blob/201911/dockers/docker-orchagent/swssconfig.sh) and written to APP DB. [CoppOrch](https://github.com/Azure/sonic-swss/blob/201911/orchagent/copporch.cpp) then translates it to Host trap/policer tables and programmed to SAI.
+
+## Proposed behaviour
+
+With the new proposal, the CoPP tables shall be loaded to Config DB instead of APP DB. The default CoPP json file shall be prebuilt to the image and loaded during initialization. Any Config DB entries present shall be configured overwriting the default CoPP tables. This also ensures backward compatibility. At a high-level, the following are the expected changes:
+
+### Schema Changes
+
+### Config DB
+
+```
+key = "COPP|name"
+name_list     = name | name,name_list
+queue         = number; strict queue priority. Higher number means higher priority.
+trap_ids      = name_list; 
+trap_action   = packet_action; trap action which will be applied to all trap_ids.
+
+;Settings for embedded policer. 
+meter_type  = "packets" | "bytes"
+mode        = "sr_tcm" | "tr_tcm" | "storm"
+color       = "aware" | "blind"
+cbs         = number ;packets or bytes depending on the meter_type value
+cir         = number ;packets or bytes depending on the meter_type value
+pbs         = number ;packets or bytes depending on the meter_type value
+pir         = number ;packets or bytes depending on the meter_type value
+
+green_action         = packet_action
+yellow_action        = packet_action
+red_action           = packet_action
+genetlink_name       = genetlink_name ;[Optional] "psample" for sFlow 
+genetlink_mcgrp_name = multicast group name; ;[Optional] "packets" for sFlow 
+```
+
+### StateDB
+
+```
+key = "COPP_TABLE|name"
+state        = "ok"
+```
+
+### coppmgr
+Introduce a *new* CoPP manager, that subscribes for the Config DB CoPP Tables and Feature Tables. Based on the feature enablement, ```coppmgr``` handles the logic to resolve whether a CoPP table shall be written to APP DB for orchagent consumption. In case if the feature requires adding only few attributes to an existing CoPP Table, ```coppmgr``` shall add the respective attributes to APP DB entry when the feature is enabled. Similar to existing swss managers, an entry with state "ok" shall be added to STATE_DB.
```

Comment:
Any application in particular that is interested in CoPP STATE_DB state?

Reply:
Not really, but just following the existing convention